### ISSUE TITLE
Custom Functions backed by algorithms

### ISSUE THREAD

**Issue by stefnotch** [stefnotch/quantum-sheet#51]
Some functions are best implemented as actual, programmatic algorithms (warning: XSS and more applies here).

For example, a number parser function `parse(0x34)` would best be done with actual code. It'd be lovely if we eventually could support defining such functions.

Relevant sympy custom functions documentation:
https://docs.sympy.org/latest/modules/functions/index.html

**Comment by phcreery** [comment]
I agree, I like the idea of providing some fundamental or sophisticated yet necessary internal functions.
